api/src/main/java/edu/wpi/first/shuffleboard/api/util/NetworkTableUtils.java

```diff
@@ -148,13 +148,16 @@ public static DataType dataTypeForEntry(String key) {
 
   /**
    * Waits for ntcore listeners to be fired. This is a <i>blocking operation</i>.
+   *
+   * @param runnable A single NetworkTable call to wait on.
    */
-  public static void waitForNtcoreEvents() {
+  public static void waitForNtcoreEvents(Runnable runnable) {
     CompletableFuture<?> future = new CompletableFuture<>();
     NetworkTablesJNI.addEntryListener("", (uid, key, value, flags) -> {
       future.complete(null);
       NetworkTablesJNI.removeEntryListener(uid);
     }, 0xFF);
+    runnable.run();
```

Comment:
re-upping with @JLLeitschuh's concern from an earlier commit:
> Couldn't the listener potentially get called before this runnable gets called? Thus it would be complete before network tables gets the value?